pkgs/applications/science/biology/slicer3d/default.nix

```diff
@@ -0,0 +1,119 @@
+{
+  stdenv,
+  lib,
+  autoPatchelfHook,
+  makeDesktopItem,
+  copyDesktopItems,
+  glib,
+  fetchzip,
+  pulseaudio,
+  xorg,
+  zlib,
+  fontconfig,
+  libxkbcommon,
+  libGL,
+  libGLU,
+  freetype,
+  cups,
+  unixODBC,
+  libxcrypt-legacy,
+  hwloc,
+  alsa-lib,
+  postgresql,
+  nss,
+  nspr,
+}:
+
+let
+  pname = "Slicer3D";
+  version = "5.6.2";
+  stability = "release";
+  os = "linux";
+  old_hwloc = hwloc.overrideAttrs (_: {
+    version = "1.4.1";
+  });
+
+  desktopItem = makeDesktopItem {
+    name = "Slicer3D";
+    exec = "SlicerApp-real";
+    desktopName = "Slicer 3D";
+    genericName = "A 3D Slicer Program"; # edit later
+    categories = [ "Graphics" ]; # Edit later
+  };
+in
+stdenv.mkDerivation {
+  inherit pname version;
+  src =
+    let
+      name = "Slicer-${version}-linux-amd64.tar.gz";
+      url = "http://download.slicer.org/download?os=${os}&stability=${stability}&version=${version}";
+    in
+    fetchzip {
+      inherit name;
+      inherit url;
+      hash = "sha256-DmJS1yrwJBcAIRnVA8VdsO4u82E1MWX+uvRC+6dEXmM=";
+      extension = "tar.gz";
+    };
+
+  desktopItems = [
+    desktopItem
+  ];
+
+  nativeBuildInputs = [
+    autoPatchelfHook
+    copyDesktopItems
+  ];
+
+  buildInputs = [
+    freetype
+    xorg.libSM
+    xorg.libICE
+    xorg.libXrender
+    xorg.libXext
+    xorg.libX11
+    zlib
+    stdenv.cc.cc.lib
+    libGL
+    libGLU
+    fontconfig
+    xorg.xkbutils
+    xorg.xcbutil
+    xorg.xcbutilwm
+    xorg.xcbutilimage
+    xorg.xcbutilrenderutil
+    xorg.xcbutilkeysyms
+    xorg.libXdamage
+    xorg.libXfixes
+    xorg.libXrandr
+    xorg.libXcursor
+    xorg.libXtst
+    xorg.libXcomposite
+    alsa-lib
+    postgresql
+    libxkbcommon
+    glib
+    pulseaudio
+    cups
+    unixODBC
+    libxcrypt-legacy
+    old_hwloc
+    nss
+    nspr
+  ];
+  autoPatchelfIgnoreMissingDeps = [ "libhwloc.so.5" ];
```

Comment:
Isn't the reason you build a custom hwloc to avoid this situation?

Reply:
Ah, i forgot to remove the custom hwloc; I was trying to get it to work at some point, but wasn't ever able to.

Reply:
I guess it's concerning that hwloc is not linked - have you used much functionality of slicer? It doesn't crash as a result of this?

Reply:
After investigating, and realizing that removing the old hwloc stopped it from building, it looks like there is just one link to an old version of that library somewhere, and wherever else it is used in the this software uses the newer version, so I think it's probably fine to leave it like this unless someone runs into an issue.

The old version is linked in `tbb`, which it looks like is prebuilt in slicer(even in the source tree) so it might be possible to replace that with what's already in nixpkgs, but I'm not quite sure if I have the time to try that right now.

Reply:
I got this fixed! I'm not sure if this was the newest version of hwloc that we could use, but it is one that works.
I do need both the current one and the overridden one since one gives `libhwloc.so.5` and the other gives `libhwloc.so.15`.
I'm not sure if there's a version with both or not.